lib/ts/MatcherUtils.cc

```diff
@@ -70,30 +72,50 @@ readIntoBuffer(const char *file_path, const char *module_name, int *read_size_pt
     return nullptr;
   }
 
-  if (file_info.st_size < 0) {
+  file_size = file_info.st_size; // number of bytes in file
+
+  if (file_size < 0) {
     Error("%s Can not get correct file size for %s file : %" PRId64 "", module_name, file_path, (int64_t)file_info.st_size);
     close(fd);
     return nullptr;
   }
+
+  ink_assert(file_size >= 0);
+  ink_assert(file_size == file_info.st_size);
+
   // Allocate a buffer large enough to hold the entire file
   //   File size should be small and this makes it easy to
   //   do two passes on the file
-  file_buf = (char *)ats_malloc(file_info.st_size + 1);
+  file_buf = (char *)ats_malloc(file_size + 1);
   // Null terminate the buffer so that string operations will work
-  file_buf[file_info.st_size] = '\0';
+  file_buf[file_size] = '\0';
+
+  ret = 0;
```

Comment:
ret should be assigned at declaration.  I would move the declaration down here.

Reply:
gotcha. updated.